devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -355,7 +355,10 @@ public void run() {
                 }
             }, "Development Mode Shutdown Hook"));
             try {
-                p.waitFor();
+                int ret = p.waitFor();
+                if (ret != 0) {
```

Comment:
This needs to be more carefully tested.
Indeed, an interrupted process may not return 0 as status.
Also, how does it behave on Windows (that's my curiosity talking)?

So basically, it needs a bit more testing.

Reply:
Yes, I can take a look later this afternoon.

Reply:
Hah, I had a patch for this but I was told at the time that the dev mode would never terminate even in case of errors, and would only exit in case of Ctrl-C so checking the return value wasn't useful.

Reply:
Yes, windows appears to show the same behavior as other platforms:

```bash
Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/arc/arc-processor/0.13.0/arc-processor-0.13.0.jar (228 kB at 311 kB/s)
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
Unrecognized option: -sfsjfj
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.609 s
[INFO] Finished at: 2019-04-09T10:06:40-07:00
[INFO] ------------------------------------------------------------------------

starksm@DESKTOP-D8936F6 MINGW64 ~/Protean/quarkus-quickstarts/using-jwt-rbac (master)
$
```

Reply:
@starksm64 and does the patch fixes the issue? Typically, I'm not sure exit codes are consistently != 0 in case of errors. But I suppose the `java` command is.

Reply:
Yes, the patch changes the behavior from:

```bash
$ ./mvnw quarkus:dev -Djvm.args="-sfsjfj"
[INFO] Scanning for projects...
[INFO]
[INFO] ----------------------< org.acme:using-jwt-rbac >-----------------------
[INFO] Building using-jwt-rbac 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- quarkus-maven-plugin:0.13.0:dev (default-cli) @ using-jwt-rbac ---
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
Unrecognized option: -sfsjfj
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.977 s
[INFO] Finished at: 2019-04-10T07:33:37-07:00
[INFO] ------------------------------------------------------------------------
```

to:
```bash
$ ./mvnw quarkus:dev -Djvm.args="-sfsjfj"
[INFO] Scanning for projects...
[INFO]
[INFO] ----------------------< org.acme:using-jwt-rbac >-----------------------
[INFO] Building using-jwt-rbac 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:dev (default-cli) @ using-jwt-rbac ---
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
Unrecognized option: -sfsjfj
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.507 s
[INFO] Finished at: 2019-04-10T07:40:39-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:dev (default-cli) on project using-jwt-rbac: Failed to run: JVM exited with error code: 1 -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

Reply:
Thanks @starksm64